src/lib/rng/stateful_rng/stateful_rng.h

```diff
@@ -72,7 +72,11 @@ class BOTAN_PUBLIC_API(2,0) Stateful_RNG : public RandomNumberGenerator
       * of the length of the input or the current seeded state of
       * the RNG.
       */
-      void initialize_with(const uint8_t input[], size_t length);
+      void initialize_with(std::span<const uint8_t> input);
+      void initialize_with(const uint8_t input[], size_t length)
+         {
+         this->initialize_with(std::span(input, length));
+         }
```

Comment:
General remark (out of scope of this pull request):

The doxygen string says:

```
      /**
      * Consume this input and mark the RNG as initialized regardless
      * of the length of the input or the current seeded state of
      * the RNG.
      */
```

Assuming that "mark the RNG as initialized" means "considering the RNG as seeded": I don't think that is the case in general!

The `input` will eventually flow down to the following code:

```C++
this->update(input);

if(8*input.size() >= security_level())
   {
   reset_reseed_counter();
   }
```

Which will reset the reseed counter only if the input was long enough. This contradicts with "regardless of the length of the input" in the doxygen comment above. I.e. `stateful_rng->initialize_with("baadcafe")` wouldn't result in a seeded `stateful_rng` object.

@randombit I think we should consider fixing this.

However: The behavior was the same before the refactoring, so I consider that out-of-scope.